happypose_ros/test/test_single_view_icp.py

```diff
@@ -0,0 +1,190 @@
+        # Clear old messages
+        self.node.clear_msg_buffer()
+
+        # Publish images several times to ensure they are received
+        start = time.time()
+        timeout = 30.0
+        ready = False
+        # Wait for the first pipeline to be triggered
+        while time.time() - start < timeout and not ready:
+            self.node.publish_image("cam_1", self.rgb, self.K, self.depth)
+            ready = proc_output.waitFor("HappyPose initialized", timeout=0.5)
+        assert ready, "Failed to trigger the pipeline!"
+
+        # Publish new to trigger parameter change
+        proc_output.assertWaitFor("Parameter change occurred", timeout=0.5)
+
+        self.node.assert_message_received("happypose/detections", timeout=20.0)
+        detections = self.node.get_received_message("happypose/detections")
+
+        self.assertEqual(
+            detections.header.frame_id, "cam_1", "Incorrect frame_id in the header!"
+        )
+        self.assertGreaterEqual(
+            len(detections.detections), 3, "Incorrect number of detected objects!"
+        )
+        self.assertEqual(
+            detections.header,
+            detections.detections[0].header,
+            "Main header differs from the detection header!",
+        )
+        self.assertEqual(
+            detections.detections[0].header,
+            detections.detections[1].header,
+            "Detected object headers differ!",
+        )
+
+        ycbv_02 = assert_and_find_detection(detections, "ycbv-obj_000002")
+        ycbv_05 = assert_and_find_detection(detections, "ycbv-obj_000005")
+        ycbv_15 = assert_and_find_detection(detections, "ycbv-obj_000015")
+
+        minimum_score = self.node.get_params(
+            ["cosypose.inference.detector.detection_th"], 5.0
+        )[0].value
+
+        self.assertGreaterEqual(ycbv_02.results[0].hypothesis.score, minimum_score)
+        self.assertLessEqual(ycbv_02.results[0].hypothesis.score, 1.0)
+
+        self.assertGreaterEqual(ycbv_05.results[0].hypothesis.score, minimum_score)
+        self.assertLessEqual(ycbv_05.results[0].hypothesis.score, 1.0)
+
+        self.assertGreaterEqual(ycbv_15.results[0].hypothesis.score, minimum_score)
+        self.assertLessEqual(ycbv_15.results[0].hypothesis.score, 1.0)
+
+        assert_pose_equal(
+            ycbv_02.results[0].pose.pose, self.ycbv_02_pose, precision=0.1
+        )
+        assert_pose_equal(
+            ycbv_05.results[0].pose.pose, self.ycbv_05_pose, precision=0.1
+        )
+        assert_pose_equal(
+            ycbv_15.results[0].pose.pose, self.ycbv_15_pose, precision=0.1
+        )
+
+        # Based on ground truth ``bbox_visib`` for image 629
+        assert_bbox(ycbv_02.bbox, [303, 34, 121, 236])
+        # Bounding box is drawn only for a half of the bottle
+        assert_bbox(ycbv_05.bbox, [394, 107, 77, 248], percent_error=50.0)
+        assert_bbox(ycbv_15.bbox, [64, 204, 267, 151])
+
+
+@pytest.mark.launch_test
```

Comment:
Could you flip the order of `generate_test_description` and class definition as it is done in multiview tests?